--- src/gtk/Makefile.in.orig	2015-08-05 23:27:22.333164538 -0700
+++ src/gtk/Makefile.in	2015-08-05 23:31:32.527888784 -0700
@@ -20,14 +20,14 @@
 
 ifeq ($(HAVE_NVCONTROL), yes)
     DEPS=$(libbackend) $(libnvcontrol)
-    LIBS=@GTK_LIBS@ -L../backend -lbackend -L../nvcontrol -lnvcontrol
+    LIBS=@GTK_LIBS@ -L../backend -lbackend -L../nvcontrol -lnvcontrol -ldl
     OBJECTS=banner.o gl.o hw.o main.o settings.o
 else
     DEPS=$(libbackend)
     LIBS=@GTK_LIBS@ -L../backend -lbackend
     OBJECTS=banner.o hw.o main.o settings.o
 endif
-LIBS += -lX11
+LIBS += -lX11 -lXext
 
 
 .PHONY: clean distclean install
