--- configure.ac.orig	2014-07-30 12:19:42.103059430 -0700
+++ configure.ac	2014-07-30 12:35:25.556636771 -0700
@@ -1096,8 +1096,10 @@
 DRI_DIRS=''
 dnl Duplicates in DRI_DIRS are removed by sorting it at the end of this block
 if test -n "$with_dri_drivers"; then
-    if test "x$enable_opengl" != xyes; then
-        AC_MSG_ERROR([--with-dri-drivers requires OpenGL])
+    if test -n "$DRI_DIRS"; then
+        if "x$enable_opengl" != xyes -a "x$enable_egl" != xyes; then
+            AC_MSG_ERROR([--with-dri-drivers requires OpenGL or EGL])
+        fi
     fi
 
     dri_drivers=`IFS=', '; echo $with_dri_drivers`
@@ -1447,11 +1449,13 @@
 AC_SUBST([LLVM_VERSION])
 AC_SUBST([CLANG_RESOURCE_DIR])
 
-case "x$enable_opengl$enable_gles1$enable_gles2" in
+case "x$enable_opengl" in
 x*yes*)
     EGL_CLIENT_APIS="$EGL_CLIENT_APIS "'$(GL_LIB)'
+    HAVE_OPENGL=yes
     ;;
 esac
+AM_CONDITIONAL(HAVE_OPENGL, test "x$HAVE_OPENGL" = xyes)
 
 AC_SUBST([VG_LIB_DEPS])
 AC_SUBST([EGL_CLIENT_APIS])
--- src/mesa/Makefile.am.orig	2014-07-30 12:36:47.225682421 -0700
+++ src/mesa/Makefile.am	2014-07-30 12:39:05.568417156 -0700
@@ -33,8 +33,10 @@
 SUBDIRS += drivers/osmesa
 endif
 
+if HAVE_OPENGL
 gldir = $(includedir)/GL
 gl_HEADERS = $(top_srcdir)/include/GL/*.h
+endif
 
 .PHONY: $(BUILDDIR)main/git_sha1.h.tmp
 $(BUILDDIR)main/git_sha1.h.tmp:
@@ -146,8 +148,10 @@
 	main/streaming-load-memcpy.c
 libmesa_sse41_la_CFLAGS = $(AM_CFLAGS) -msse4.1
 
+if HAVE_OPENGL
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gl.pc
+endif
 
 $(BUILDDIR)program/lex.yy.c: program/program_lexer.l
 	$(AM_V_at)$(MKDIR_P) program
