--- Makefile.orig	2015-04-04 20:44:55.422234601 -0700
+++ Makefile	2015-04-04 20:47:05.205580882 -0700
@@ -512,16 +512,6 @@
 
 ALL_OBJS += ${CGPT_OBJS}
 
-CGPT_WRAPPER = ${BUILD}/cgpt/cgpt_wrapper
-
-CGPT_WRAPPER_SRCS = \
-	cgpt/cgpt_nor.c \
-	cgpt/cgpt_wrapper.c
-
-CGPT_WRAPPER_OBJS = ${CGPT_WRAPPER_SRCS:%.c=${BUILD}/%.o}
-
-ALL_OBJS += ${CGPT_WRAPPER_OBJS}
-
 # Utility defaults
 UTIL_DEFAULTS = ${BUILD}/default/vboot_reference
 
@@ -771,7 +761,7 @@
 	pc_files_install
 
 .PHONY: install_mtd
-install_mtd: install cgpt_wrapper_install
+install_mtd: install
 
 .PHONY: install_for_test
 install_for_test: override DESTDIR = ${TEST_INSTALL_DIR}
@@ -948,15 +938,8 @@
 # ----------------------------------------------------------------------------
 # CGPT library and utility
 
-.PHONY: cgpt_wrapper
-cgpt_wrapper: ${CGPT_WRAPPER}
-
-${CGPT_WRAPPER}: ${CGPT_WRAPPER_OBJS} ${UTILLIB}
-	@$(PRINTF) "    LD            $(subst ${BUILD}/,,$@)\n"
-	${Q}${LD} -o ${CGPT_WRAPPER} ${CFLAGS} $^
-
 .PHONY: cgpt
-cgpt: ${CGPT} ${CGPT_WRAPPER}
+cgpt: ${CGPT}
 
 ${CGPT}: LDLIBS += -luuid
 
@@ -970,15 +953,6 @@
 	${Q}mkdir -p ${UB_DIR}
 	${Q}${INSTALL} -t ${UB_DIR} $^
 
-.PHONY: cgpt_wrapper_install
-cgpt_wrapper_install: cgpt_install ${CGPT_WRAPPER}
-	@$(PRINTF) "    INSTALL       cgpt_wrapper\n"
-	${Q}${INSTALL} -t ${UB_DIR} ${CGPT_WRAPPER}
-	${Q}mv ${UB_DIR}/$(notdir ${CGPT}) \
-		${UB_DIR}/$(notdir ${CGPT}).bin
-	${Q}mv ${UB_DIR}/$(notdir ${CGPT_WRAPPER}) \
-		${UB_DIR}/$(notdir ${CGPT})
-
 # ----------------------------------------------------------------------------
 # Utilities
 
