--- tools/clang/lib/Driver/Tools.cpp.orig	2015-07-25 22:17:56.489628405 -0700
+++ tools/clang/lib/Driver/Tools.cpp	2015-07-25 22:21:43.173439848 -0700
@@ -1540,7 +1540,7 @@
   // forward.
   CmdArgs.push_back("-plugin");
   std::string Plugin =
-      ToolChain.getDriver().Dir + "/../lib" CLANG_LIBDIR_SUFFIX "/LLVMgold.so";
+      ToolChain.getDriver().Dir + "/../@libdir@/LLVMgold.so";
   CmdArgs.push_back(Args.MakeArgString(Plugin));
 
   // Try to pass driver level flags relevant to LTO code generation down to
--- tools/clang/tools/scan-build/scan-build.orig	2015-07-25 22:21:59.843965110 -0700
+++ tools/clang/tools/scan-build/scan-build	2015-07-25 22:23:23.193258335 -0700
@@ -437,7 +437,7 @@
 
   my $Dir = shift;
 
-  my $JS = Cwd::realpath("$RealBin/sorttable.js");
+  my $JS = Cwd::realpath("@EPREFIX@/usr/share/llvm/sorttable.js");
 
   DieDiag("Cannot find 'sorttable.js'.\n")
     if (! -r $JS);
@@ -447,7 +447,7 @@
   DieDiag("Could not copy 'sorttable.js' to '$Dir'.\n")
     if (! -r "$Dir/sorttable.js");
 
-  my $CSS = Cwd::realpath("$RealBin/scanview.css");
+  my $CSS = Cwd::realpath("@EPREFIX@/usr/share/llvm/scanview.css");
 
   DieDiag("Cannot find 'scanview.css'.\n")
     if (! -r $CSS);
--- tools/clang/tools/scan-view/scan-view.orig	2015-07-25 22:23:50.107439931 -0700
+++ tools/clang/tools/scan-view/scan-view	2015-07-25 22:24:46.702557184 -0700
@@ -34,7 +34,7 @@
     import urllib, webbrowser
 
     url = 'http://%s:%d'%(options.host, port)
-    
+
     # Wait for server to start...
     if options.debug:
         sys.stderr.write('%s: Waiting for server.' % sys.argv[0])
@@ -54,7 +54,7 @@
     webbrowser.open(url)
 
 def run(port, options, root):
-    import ScanView
+    from clang import ScanView
     try:
         print 'Starting scan-view at: http://%s:%d'%(options.host,
                                                      port)
