diff --git a/common/Makefile.am b/common/Makefile.am
index 1e794e4..6d3490e 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -14,7 +14,6 @@
 # Turn off -Wshadow - Xorg headers seem to declare a lot of globals
 # which can conflict - index, range, etc.
 AM_CFLAGS = $(filter-out -Wnested-externs -Wcast-qual -Wredundant-decls \
-	-Werror=write-strings -Wshadow,$(CWARNFLAGS)) \
 	$(XORG_CFLAGS)
 
 noinst_LTLIBRARIES = libcommon.la
diff --git a/etnaviv/Makefile.am b/etnaviv/Makefile.am
index 95e3775..3f197a7 100644
--- a/etnaviv/Makefile.am
+++ b/etnaviv/Makefile.am
@@ -14,7 +14,6 @@
 # Turn off -Wshadow - Xorg headers seem to declare a lot of globals
 # which can conflict - index, range, etc.
 AM_CFLAGS = $(filter-out -Wnested-externs -Wcast-qual -Wredundant-decls \
-	-Werror=write-strings -Wshadow,$(CWARNFLAGS)) \
 	$(XORG_CFLAGS) $(ETNAVIV_CFLAGS) $(DRMARMADA_CFLAGS) $(UDEV_CFLAGS) \
 	-I$(top_srcdir)/src -I$(top_srcdir)/common
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 3afed38..75650c3 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -14,7 +14,6 @@
 # Turn off -Wshadow - Xorg headers seem to declare a lot of globals
 # which can conflict - index, range, etc.
 AM_CFLAGS = $(filter-out -Wnested-externs -Wcast-qual -Wredundant-decls \
-	-Werror=write-strings -Wshadow,$(CWARNFLAGS)) \
 	$(XORG_CFLAGS) $(DRMARMADA_CFLAGS) $(UDEV_CFLAGS) \
 	-I$(top_srcdir)/common
 
diff --git a/vivante/Makefile.am b/vivante/Makefile.am
index 686dd85..1e6c66f 100644
--- a/vivante/Makefile.am
+++ b/vivante/Makefile.am
@@ -14,7 +14,6 @@
 # Turn off -Wshadow - Xorg headers seem to declare a lot of globals
 # which can conflict - index, range, etc.
 AM_CFLAGS = $(filter-out -Wnested-externs -Wcast-qual -Wredundant-decls \
-	-Werror=write-strings -Wshadow,$(CWARNFLAGS)) \
 	$(XORG_CFLAGS) $(LIBGAL_CFLAGS) $(DRMARMADA_CFLAGS) $(UDEV_CFLAGS) \
 	-I$(top_srcdir)/src -I$(top_srcdir)/common
 
