--- a/src/drm/cairo-drm-intel-private.h	2015-08-11 12:44:31.602456665 -0700
+++ b/src/drm/cairo-drm-intel-private.h	2015-08-11 12:45:41.894676260 -0700
@@ -39,6 +39,7 @@
 #include "cairo-mutex-private.h"
 #include "cairo-rtree-private.h"
 #include "cairo-types-private.h"
+#include "cairo-pattern-private.h"
 
 #include "cairo-drm-intel-ioctl-private.h"
 
diff --git a/src/drm/cairo-drm-private.h b/src/drm/cairo-drm-private.h
index 2db7f38..1a66a78 100644
--- a/src/drm/cairo-drm-private.h
+++ b/src/drm/cairo-drm-private.h
@@ -43,6 +43,8 @@
 #include "cairo-surface-private.h"
 
 #include <sys/types.h> /* dev_t */
+#include <sys/stat.h>  /* fstat */
+#include <unistd.h>
 
 typedef struct _cairo_drm_device cairo_drm_device_t;
 
