--- src/cairo-gl-private.h.orig	2014-07-30 15:54:46.365104949 -0700
+++ src/cairo-gl-private.h	2014-07-30 15:59:26.936115538 -0700
@@ -61,8 +61,8 @@
 #include <assert.h>
 
 #if CAIRO_HAS_GL_SURFACE
+#define GL_GLEXT_LEGACY
 #include <GL/gl.h>
-#include <GL/glext.h>
 #elif CAIRO_HAS_GLESV2_SURFACE
 #include <GLES2/gl2.h>
 #include <GLES2/gl2ext.h>
--- src/cairo-gl-gradient-private.h.orig	2014-07-30 16:05:59.564379463 -0700
+++ src/cairo-gl-gradient-private.h	2014-07-30 16:08:11.223801733 -0700
@@ -52,8 +52,8 @@
 #include "cairo-gl.h"
 
 #if CAIRO_HAS_GL_SURFACE
+#define GL_GLEXT_LEGACY
 #include <GL/gl.h>
-#include <GL/glext.h>
 #elif CAIRO_HAS_GLESV2_SURFACE
 #include <GLES2/gl2.h>
 #include <GLES2/gl2ext.h>
--- src/cairo-vg-surface.c.orig	2014-07-30 16:11:23.553224591 -0700
+++ src/cairo-vg-surface.c	2014-07-30 16:14:42.215639129 -0700
@@ -1675,6 +1675,7 @@
 };
 
 #if CAIRO_HAS_GLX_FUNCTIONS
+#define GLX_GLXEXT_LEGACY
 #include <GL/glx.h>
 
 static cairo_status_t
--- src/cairo-gl.h.orig	2014-07-30 16:16:49.265326332 -0700
+++ src/cairo-gl.h	2014-07-30 16:20:14.224072273 -0700
@@ -93,6 +93,7 @@
 				  cairo_bool_t		 thread_aware);
 
 #if CAIRO_HAS_GLX_FUNCTIONS
+#define GLX_GLXEXT_LEGACY
 #include <GL/glx.h>
 
 cairo_public cairo_device_t *
