--- Makefile.orig	2015-08-02 13:43:21.793770111 -0700
+++ Makefile	2015-08-02 13:44:46.390900498 -0700
@@ -1,7 +1,13 @@
+diag:
+	$(MAKE) -C pub/diag
+
 all: .patch
 	$(MAKE) -C pub/diag
 	$(MAKE) -C pub/isa-diag
 
+install-diag:
+	$(MAKE) -C pub/diag install
+
 install:
 	$(MAKE) -C pub/diag $@
 	$(MAKE) -C pub/isa-diag $@
@@ -21,4 +27,4 @@
 dist:
 	./make-dist.sh $(GV)
 
-.PHONY: all clean dist install unpatch
+.PHONY: all clean dist install unpatch diag install-diag
