From: Jim Chen <nchen@mozilla.com>
Date: Mon, 1 May 2017 14:46:00 -0400
Subject: Bug 1357874 - Define GETRANDOM_NR macro for AArch64. r=luke

---
 js/src/jsmath.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/src/jsmath.cpp b/js/src/jsmath.cpp
index a43651fbb834..0f8c6f4808cf 100644
--- a/js/src/jsmath.cpp
+++ b/js/src/jsmath.cpp
@@ -66,6 +66,8 @@
 #  define GETRANDOM_NR 318
 # elif defined(__i386__)
 #  define GETRANDOM_NR 355
+# elif defined(__aarch64__)
+#  define GETRANDOM_NR 278
 # elif defined(__arm__)
 #  define GETRANDOM_NR 384
 // Added other architectures:
