--- granite-0.3.0/lib/Widgets/ContractorView.vala	2014-05-25 21:42:24.000000000 +0200
+++ granite-0.3.0/lib/Widgets/ContractorView.vala	2017-09-30 18:19:03.980547473 +0200
@@ -181,7 +181,7 @@ 
 	        string text = value.get_string ();
 	        
 	        if (text[3:text.index_of ("</b>")] in names) {
-	            list.remove (it);
+	            list.remove (ref it);
 	            blacklisted_pos += cur_pos;
             }
 	        if (!check)

