--- a/src/client/Makefile	2017-11-29 17:18:41.718068820 -0800
+++ b/src/client/Makefile	2017-11-29 17:24:24.834304096 -0800
@@ -44,6 +44,8 @@
 SED           = sed
 STRIP         = armv7a-hardfloat-linux-gnueabi-strip
 
+LINK_FIX_FLAGS = -Wl,-unresolved-symbols=ignore-in-shared-libs
+
 ####### Output directory
 
 OBJECTS_DIR   = .obj/
@@ -376,7 +378,7 @@
 ../../lib/$(TARGET): /usr/lib/libQt5FontDatabaseSupport.a /usr/lib/libQt5EventDispatcherSupport.a /usr/lib/libQt5ThemeSupport.a /usr/lib/libQt5ServiceSupport.a $(OBJECTS) $(SUBLIBS) $(OBJCOMP)  
 	@test -d ../../lib/ || mkdir -p ../../lib/
 	-$(DEL_FILE) $(TARGET) $(TARGET0) $(TARGET1) $(TARGET2)
-	$(LINK) $(LFLAGS) -o $(TARGET) $(OBJECTS) $(LIBS) $(OBJCOMP)
+	$(LINK) $(LFLAGS) -o $(TARGET) $(OBJECTS) $(LIBS) $(LINK_FIX_FLAGS) $(OBJCOMP)
 	-ln -s $(TARGET) $(TARGET0)
 	-ln -s $(TARGET) $(TARGET1)
 	-ln -s $(TARGET) $(TARGET2)
