--- src/drmmode_display.c.orig	2014-01-12 13:05:59.480000000 -0800
+++ src/drmmode_display.c	2014-01-12 13:15:06.560000000 -0800
@@ -461,8 +461,13 @@
         }
 
         if (drmmode_crtc->slave_damage) {
+#if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,14,99,2,0)
+            DamageUnregister(drmmode_crtc->slave_damage);
+#else
             DamageUnregister(&crtc->randr_crtc->scanout_pixmap->drawable,
                              drmmode_crtc->slave_damage);
+#endif
+            DamageDestroy(drmmode_crtc->slave_damage);
             drmmode_crtc->slave_damage = NULL;
         }
 
--- src/driver.c.orig	2014-01-12 13:26:13.410000000 -0800
+++ src/driver.c	2014-01-12 13:28:22.910000000 -0800
@@ -128,8 +128,12 @@
     ret = dispatch_dirty_region(scrn, pixmap, tegra->damage, fb_id);
     if (ret == -EINVAL || ret == -ENOSYS) {
         tegra->dirty_enabled = FALSE;
+#if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,14,99,2,0)
+        DamageUnregister(tegra->damage);
+#else
         DamageUnregister(&pScreen->GetScreenPixmap(pScreen)->drawable,
                          tegra->damage);
+#endif
         DamageDestroy(tegra->damage);
         tegra->damage = NULL;
         xf86DrvMsg(scrn->scrnIndex, X_INFO, "Disabling kernel dirty updates, not required.\n");
--- src/driver.c.orig	2014-01-12 13:55:18.210000000 -0800
+++ src/driver.c	2014-01-12 13:57:40.270000000 -0800
@@ -570,8 +570,12 @@
     TegraPtr tegra = TegraPTR(pScrn);
 
     if (tegra->damage) {
+#if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,14,99,2,0)
+        DamageUnregister(tegra->damage);
+#else
         DamageUnregister(&pScreen->GetScreenPixmap(pScreen)->drawable,
                          tegra->damage);
+#endif
         DamageDestroy(tegra->damage);
         tegra->damage = NULL;
     }
