--- a/media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features.cc	
+++ a/media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features.cc	
@@ -61,12 +61,25 @@ static int GetCPUInfo(CPUFeature feature
   return 0;
 }
 #else
 // Default to straight C for other platforms.
 static int GetCPUInfo(CPUFeature feature) {
   (void)feature;
   return 0;
 }
+
+#if !defined(WEBRTC_GONK) && !defined(ANDROID)
+#ifdef WEBRTC_ARCH_ARM_V7
+uint64_t WebRtc_GetCPUFeaturesARM(void) {
+  return kCPUFeatureARMv7
+#ifdef WEBRTC_ARCH_ARM_NEON
+         | kCPUFeatureNEON
+#endif
+         | kCPUFeatureVFPv3;
+}
+#endif // WEBRTC_ARCH_ARM_V7
+#endif // !WEBRTC_GONK && !ANDROID
+
 #endif
 
 WebRtc_CPUInfo WebRtc_GetCPUInfo = GetCPUInfo;
 WebRtc_CPUInfo WebRtc_GetCPUInfoNoASM = GetCPUInfoNoASM;
