--- configure.orig	2015-04-22 14:19:27.725026443 -0700
+++ configure	2015-04-22 14:31:32.890026789 -0700
@@ -6765,6 +6765,13 @@
         MOZ_FLOAT_ABI=softfp
         MOZ_ALIGN=no
         ;;
+    arm-Linux)
+        MOZ_THUMB=yes
+        MOZ_ARCH=armv7-a
+        MOZ_FPU=neon
+        MOZ_FLOAT_ABI=hard
+        MOZ_SOFT_FLOAT=no
+        ;;
     arm-Darwin)
         MOZ_ARCH=toolchain-default
         MOZ_THUMB=yes
