--- Source/WebCore/Modules/webaudio/AudioBuffer.cpp.orig	2014-07-23 18:25:12.928403571 -0700
+++ Source/WebCore/Modules/webaudio/AudioBuffer.cpp	2014-07-23 18:26:04.782192434 -0700
@@ -38,6 +38,9 @@
 #include "ExceptionCode.h"
 #include "ExceptionCodePlaceholder.h"
 
+#include <runtime/JSCInlines.h> 
+#include <runtime/TypedArrayInlines.h>
+
 namespace WebCore {
 
 PassRefPtr<AudioBuffer> AudioBuffer::create(unsigned numberOfChannels, size_t numberOfFrames, float sampleRate)
