--- Source/WebCore/html/canvas/WebGLGetInfo.h.orig	2014-07-25 13:18:07.321550275 -0700
+++ Source/WebCore/html/canvas/WebGLGetInfo.h	2014-07-25 13:17:35.146585445 -0700
@@ -37,6 +37,7 @@
 #include <runtime/Int32Array.h>
 #include <runtime/Uint32Array.h>
 #include <runtime/Uint8Array.h>
+#include <runtime/TypedArrayInlines.h>
 #include <wtf/PassRefPtr.h>
 #include <wtf/RefPtr.h>
 #include <wtf/text/WTFString.h>
--- Source/WebCore/Modules/webaudio/AudioParamTimeline.h.orig	2014-07-25 13:16:04.101400815 -0700
+++ Source/WebCore/Modules/webaudio/AudioParamTimeline.h	2014-07-25 13:16:20.757186593 -0700
@@ -32,6 +32,7 @@
 #include "AudioContext.h"
 #include <mutex>
 #include <runtime/Float32Array.h>
+#include <runtime/TypedArrayInlines.h>
 #include <wtf/PassRefPtr.h>
 #include <wtf/RefCounted.h>
 #include <wtf/Vector.h>
--- Source/WebCore/Modules/webaudio/AudioBuffer.h.orig	2014-07-25 13:16:57.955219213 -0700
+++ Source/WebCore/Modules/webaudio/AudioBuffer.h	2014-07-25 14:43:28.460321583 -0700
@@ -30,6 +30,7 @@
 #define AudioBuffer_h
 
 #include <runtime/Float32Array.h>
+#include <runtime/TypedArrayInlines.h>
 #include <wtf/PassRefPtr.h>
 #include <wtf/RefCounted.h>
 #include <wtf/RefPtr.h>
