--- CMakeLists.txt.orig	2015-12-06 21:29:59.953920054 -0800
+++ CMakeLists.txt	2015-12-06 21:31:49.780764856 -0800
@@ -116,6 +116,7 @@
 else ()
     message(FATAL_ERROR "Unknown CPU '${LOWERCASE_CMAKE_SYSTEM_PROCESSOR}'")
 endif ()
+message(STATUS "Detected CPU: '${LOWERCASE_CMAKE_SYSTEM_PROCESSOR}'")
 
 # -----------------------------------------------------------------------------
 # Determine the operating system
