--- a/Source/cmake/WebKitCommon.cmake	2017-11-20 11:00:46.010738308 -0800
+++ b/Source/cmake/WebKitCommon.cmake	2017-11-20 12:07:36.479693880 -0800
@@ -27,9 +27,9 @@
     # We cannot check for RUBY_FOUND because it is set only when the full package is installed and
     # the only thing we need is the interpreter. Unlike Python, cmake does not provide a macro
     # for finding only the Ruby interpreter.
-    find_package(Ruby 1.9)
-    if (NOT RUBY_EXECUTABLE OR RUBY_VERSION VERSION_LESS 1.9)
-        message(FATAL_ERROR "Ruby 1.9 or higher is required.")
+    find_package(Ruby 1.8)
+    if (NOT RUBY_EXECUTABLE OR RUBY_VERSION VERSION_LESS 1.8)
+        message(FATAL_ERROR "Ruby 1.8 or higher is required.")
     endif ()
 
     # -----------------------------------------------------------------------------
