--- a/Source/WTF/wtf/URLHelpers.cpp	2019-11-07 17:11:13.419899196 -0800
+++ b/Source/WTF/wtf/URLHelpers.cpp	2019-11-07 17:12:25.762788750 -0800
@@ -298,7 +298,7 @@
     Optional<UChar32> previousCodePoint;
     while (i < length) {
         UChar32 c;
-        U16_NEXT(buffer, i, length, c)
+        U16_NEXT(buffer, i, length, c);
         UErrorCode error = U_ZERO_ERROR;
         UScriptCode script = uscript_getScript(c, &error);
         if (error != U_ZERO_ERROR) {
--- a/Source/WebCore/dom/Document.cpp	2019-11-08 00:55:19.932462186 -0800
+++ b/Source/WebCore/dom/Document.cpp	2019-11-08 00:56:48.491102833 -0800
@@ -5072,12 +5072,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -5137,7 +5137,7 @@
 
     for (unsigned i = 0; i < length; ) {
         UChar32 c;
-        U16_NEXT(qualifiedName, i, length, c)
+        U16_NEXT(qualifiedName, i, length, c);
         if (c == ':') {
             if (sawColon)
                 return Exception { InvalidCharacterError };
